src/Google/GlobalSiteTag.php

```diff
@@ -109,6 +138,73 @@ function () {
 				$this->display_page_view_event_snippet();
 			}
 		);
+
+		$this->product_data_hooks();
+		$this->register_assets();
+	}
+
+	/**
+	 * Attach filters to add product data required for tracking events.
+	 */
+	protected function product_data_hooks() {
+		// Add product data for any add_to_cart link.
+		add_filter(
+			'woocommerce_loop_add_to_cart_link',
+			function ( $link, $product, $args ) {
+				$this->add_product_data( $product );
+				return $link;
+			},
+			10,
+			3
+		);
+
+		// Add display name for an available variation.
+		add_filter(
+			'woocommerce_available_variation',
+			function ( $data, $instance, $variation ) {
+				$data['display_name'] = $variation->get_name();
+				return $data;
+			},
+			10,
+			3
+		);
+	}
+
+	/**
+	 * Register and enqueue assets for gtag events in blocks.
+	 */
+	protected function register_assets() {
+		$gtag_events = new ScriptWithBuiltDependenciesAsset(
+			'gla-gtag-events',
+			'js/build/gtag-events',
+			"{$this->get_root_dir()}/js/build/gtag-events.asset.php",
+			new BuiltScriptDependencyArray(
```

Comment:
💅  Outside scope of this PR but this name is not accurate since we are loading Deps and Version here, not only deps.

Reply:
In my opinion that would depend on how you look at dependencies. If you see a single dependency as a script + version, then a DependencyArray would be the right name for it.

Reply:
For me one thing is the version of the script and other the dependencies required by that script. But checking the codebase it's how WP defines this... so I think it doesn't worth it the update